File: algorithms/0504.base-7/solution.go
func convertToBase7(num int) string {
	if num == 0 {
		return "0"
	}

	var res string
	var n int
	if num > 0 {
		n = num
	} else {
		n = num * -1
	}

	for n > 0 {
		rem := n % 7
		n /= 7
		res = strconv.Itoa(rem) + res
	}

	if num > 0 {
		return res
	} else {
		return "-" + res
	}
}

File: algorithms/0504.base-7/solution.rb
# @param {Integer} num
# @return {String}
def convert_to_base7(num)
  ## Lazy solution
  # num.to_s(7)

  return "0" if num == 0

  res = []
  n = num.abs

  while n > 0
    rem = n % 7
    n /= 7
    res.unshift(rem.to_s)
  end

  res.unshift("-") if num < 0
  res.join
end

